frontend/packages/rhoas-plugin/src/components/streams-list/StreamsInstanceTable.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import {
+  sortable,
+  cellWidth,
+  Table,
+  TableHeader,
+  TableBody,
+  RowSelectVariant
+} from '@patternfly/react-table';
+import { Timestamp } from '@console/internal/components/utils';
+import './StreamsInstanceTable.css';
+import { useTranslation } from 'react-i18next';
+
+type FormattedKafkas = {
+  cells: JSX.Element[];
+  selected: boolean;
+};
+
+const StreamsInstanceTable: any = ({ kafkaArray, setSelectedKafka, currentKafkaConnections, setAllKafkasConnected }) => {
+
+  const [formattedKafkas, setFormattedKafkas] = React.useState<FormattedKafkas[]>([]);
+  const { t } = useTranslation();
+
+  const formatTableRowData = () => {
+    const tableRow = [];
+    kafkaArray.forEach(row => {
+      const { name, bootstrapServerHost, provider, region, owner, createdAt } = row;
+      tableRow.push({
+        cells: [
+          { title: name },
+          { title: <a href="/">{bootstrapServerHost}</a> },
+          { title: provider },
+          { title: region },
+          { title: <a href="/">{owner}</a> },
+          { title: <Timestamp timestamp={createdAt} /> },
+        ]
+      });
+    });
+
+    kafkaArray.forEach((kafka, index) => {
+      if (currentKafkaConnections.includes(kafka.id)) {
+        tableRow[index].disableSelection = true;
+      }
+    })
```

Comment:
Can we make use of map as below

```suggestion
   const tableRow  = kafkaArray.map(({ kafka , name, bootstrapServerHost, provider, region, owner, createdAt }) => {
   return {
        cells: [
          { title: name },
          { title: <a href="/">{bootstrapServerHost}</a> },
          { title: provider },
          { title: region },
          { title: <a href="/">{owner}</a> },
          { title: <Timestamp timestamp={createdAt} /> },
        ],
       ...(currentKafkaConnections.includes(kafka.id) && { disableSelection : true }
      }
})
```

Reply:
+1